apps/air-discount-scheme/backend/src/app/modules/discount/discount.service.ts

```diff
@@ -75,10 +92,39 @@ export class DiscountService {
     return new Discount(discountCode, cacheValue.nationalId, ttl)
   }
 
-  async useDiscount(discountCode: string, nationalId: string): Promise<void> {
+  async useDiscount(
+    discountCode: string,
+    nationalId: string,
+    flightId: string,
+  ): Promise<void> {
     const discountCodeCacheKey = this.getDiscountCodeCacheKey(discountCode)
     const nationalIdCacheKey = this.getNationalIdCacheKey(nationalId)
+    const cacheId = await this.cacheManager.get(discountCodeCacheKey)
     await this.cacheManager.del(discountCodeCacheKey)
     await this.cacheManager.del(nationalIdCacheKey)
+
+    const usedCacheKey = this.getUsedDiscountCodeCacheKey(flightId)
+    await this.cacheManager.set(usedCacheKey, cacheId, { ttl: ONE_DAY })
```

Comment:
`ttl` needs to be the `ttl` of the `cacheId` key. Otherwise they'll get out of sync.
```typescript
const cacheId = await this.cacheManager.get(discountCodeCacheKey)
const ttl = await this.cacheManager.ttl(cacheId)

await this.cacheManager.del(discountCodeCacheKey)
await this.cacheManager.del(nationalIdCacheKey)

const usedCacheKey = this.getUsedDiscountCodeCacheKey(flightId)
await this.cacheManager.set(usedCacheKey, cacheId, { ttl })
```